src/RateLimit.js

```diff
@@ -193,11 +193,11 @@ class RateLimit {
     }
 
     _isWhitelisted(key) {
-        const arr = key.split('::');
+        const arr = key.split('|');
```

Comment:
create a separator parameter defaulting to `'::'`

Reply:
another alternative would be to create a new function that takes a function as it's parameter, allowing full customization of the key parsing logic

Reply:
Good point regarding the backward compatibility. 
I'll do it how you suggested.

However from a new user perspective, it is not obvious why this separator have to be adjusted to `'|'`,  if `userIds` are used in the whitelist option. 

Changing the default separator to `'|'` would be more inline with default `keyGenerator` that creates the key by using this separator: <prefix>|<userId or ip>. 

How about to change the default separator to '|' in a future release and now  just displaying a deprecation warning that key extraction behavior will be changed?